### PR TITLE
fix(retainer): show `$SYS` topics when iterating over all topics

### PR DESCRIPTION
**Pull request by thalesmg**
Fixes https://emqx.atlassian.net/browse/EMQX-13399

Release version: v/e5.8.2

## Summary

Previously, we used to show such topics in the API.  Alternatively, we would need special handling when computing the total number of retained messages when returning the total count in the API, if we were to start hiding those.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

**Review by keynslug — COMMENTED** [review]
Thanks for taking care of that. Should have thought about this outcome. 👍🏼

File: apps/emqx_retainer/test/emqx_retainer_api_SUITE.erl [hunk 1]
```diff
@@ -424,6 +424,21 @@ t_match_and_clean(Config) ->
     {ok, LookupJson2} = request_api(get, API),
     ?assertMatch(#{data := []}, decode_json(LookupJson2)).
 
+%% Checks that we can see `$SYS' messages in the API.
+%% https://emqx.atlassian.net/browse/EMQX-13399
+t_retained_sys_messages(_Config) ->
+    ok = emqx_sys:populate_sys_topics(),
```
[the remark this answers is not in the document]
Comment (thalesmg) [reply apps/emqx_retainer/test/emqx_retainer_api_SUITE.erl]: :checkered_flag:

File: apps/emqx_retainer/src/emqx_retainer_mnesia.erl [hunk 2]
```diff
@@ -310,7 +310,7 @@ page_read(_State, Topic, Deadline, Page, Limit) ->
     S0 =
         case Topic of
             undefined ->
-                msg_stream(search_stream(undefined, ['#'], Deadline));
+                msg_stream(search_stream(undefined, all, Deadline));
```
[the remark this answers is not in the document]
Comment (thalesmg) [reply apps/emqx_retainer/src/emqx_retainer_mnesia.erl]: :checkered_flag:

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-4923)

**Pull request merged**